src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -0,0 +1,829 @@
+        class="k-code-block-search-error"
+      >
+        {{ regExpError.message }}
+      </p>
+
+      <p
+        v-else-if="!isProcessing && query !== ''"
+        class="k-code-block-search-results"
+      >
+        <template v-if="!isShowingFilteredCode && numberOfMatches > 0 && typeof currentLineIndex === 'number'">
+          {{ currentLineIndex + 1 }} of
+        </template>
+
+        {{ numberOfMatches }} results
+      </p>
+
+      <KIcon
+        v-else
+        class="k-is-processing-icon"
+        :class="{ 'k-is-processing-icon--is-visible': isProcessing }"
+        icon="spinner"
+        data-testid="k-is-processing-icon"
+      />
+
+      <KLabel
+        :for="`${props.id}-search-input`"
+        class="k-code-block-search"
+      >
+        <span class="k-visually-hidden">Highlight</span>
+
+        <KInput
+          :id="`${props.id}-search-input`"
+          class="k-code-block-search__input"
+          type="text"
+          :model-value="query"
+          data-testid="k-code-block-search-input"
+          @input="handleSearch"
+        />
+      </KLabel>
+
+      <KButton
+        class="k-regexp-mode-button"
+        type="button"
+        :appearance="regExpButtonAppearance"
+        :aria-pressed="isRegExpMode"
+        :is-rounded="false"
+        size="small"
+        title="Use regular expression (Alt+R)"
+        data-testid="k-regexp-mode-button"
+        @click="toggleRegExpMode"
+      >
+        <span class="k-visually-hidden">RegExp mode enabled</span>
+
+        .*
+      </KButton>
+
+      <KButton
+        class="k-filter-mode-button"
+        type="button"
+        icon="filter"
+        :appearance="isFilterMode ? 'secondary' : 'outline'"
+        :aria-pressed="isFilterMode"
+        :is-rounded="false"
+        size="small"
+        title="Filter results (Alt+F)"
+        data-testid="k-filter-mode-button"
+        @click="toggleFilterMode"
+      >
+        <template #icon>
+          <KIcon
+            class="k-button-icon"
+            icon="filter"
+            size="16"
+            title="Filter results (Alt+F)"
+            color="currentColor"
+          />
+        </template>
+
+        <span class="k-visually-hidden">Filter mode enabled</span>
+      </KButton>
+
+      <KButton
+        class="k-previous-match-button"
+        type="button"
+        :is-rounded="false"
+        size="small"
+        title="Previous match (Shift+Enter)"
+        :disabled="matchingLineNumbers.length === 0 || isFilterMode"
+        data-testid="k-previous-match-button"
+        @click="jumpToPreviousMatch"
+      >
+        <template #icon>
+          <KIcon
+            class="k-button-icon"
+            icon="chevronUp"
+            size="16"
+            title="Previous match (Shift+Enter)"
+            color="currentColor"
+          />
+        </template>
+
+        <span class="k-visually-hidden">Previous match</span>
+      </KButton>
+
+      <KButton
+        class="k-next-match-button"
```

Comment:
The classes and test ids in the file:

Instead of `k-{random}` can we prefix with `k-code-block-*`?

This allows for dynamic selector matching namespaced to the component.

Comment applies to all random classes/testid attributes that are using `k-*`

Reply:
Sure, I changed them.